### ISSUE TITLE
Narrator does not annouce HyperLink in reverse (Shift+Tab) on FlowDocumentScrollViewer

### ISSUE THREAD

**Issue by grubioe** [dotnet/wpf#790]
Copied from VSO ID: 608596

Does the bug reproduce also in WPF for .NET Framework 4.8?: Yes

Simple WPF program attached to demonstrate the issue

[Sample.zip](https://github.com/dotnet/wpf/files/3237805/Sample.zip)
 
 **Problem description:**
Narrator does not annouce HyperLink in reverse (Shift+Tab) on FlowDocumentScrollViewer
 
 **Repro:**
 1.Tab to the FlowDocument and Narrator will announce "Checked out branch Ham, which is out of date. [Pull]() the latest changes to update.",
2. Tab.  Focus is now on the hyperlink and it correctly says "Pull hyperlink"
3. Tab again (focus will be on the Textblock's hyperlink)
4. Shift Tab.  Focus is now on the hyperlink but it says "Checked out branch Ham, which is out of date. [Pull]() the latest changes to update."
5. Shift Tab. Focus is now on the FlowDocument and it will announce "Checked out branch Ham, which is out of date. [Pull]() the latest changes to update.",

Step 4 should announce the same as Step 2.  Going forward or backwards shouldn't change what narrator reads.

I attached the program for easy replay, but here is the crux of the XAML.  Only changes from default WPF project are in the MainWindow.xaml file.

    <FlowDocumentScrollViewer Grid.Column="1" AutomationProperties.Name="Checked out branch Ham, which is out of date. [Pull]() the latest changes to update.">
            <FlowDocument>
                <Paragraph>
                    <Run>Checked out branch Ham, which is out of date</Run>
                    <Hyperlink NavigateUri="http://www.msn.com">Pull</Hyperlink>
                    <Run>the latest changes to update.</Run>
                </Paragraph>
            </FlowDocument>               
        </FlowDocumentScrollViewer>

**Comment by anjalisheel-wpf** [comment]
@grubioe 

The problem is that Narrator announces one thing after a Tab navigation, and a different thing after a Shift-Tab navigation (to the same hyperlink).  Tab is reading the hyperlink’s Name, while Shift-Tab is reading the FlowDocumentScrollViewer’s Name.

When we call AutomationPeer.GetName, it sees a call for the focused element (the hyperlink), that should return its text “Pull”.

The symptoms in the bug sound more like focus is getting set to an unexpected element, or that perhaps Narrator is asking for the Name of the wrong element. 

This is working as per the design, hence closing it.
